agent/agent.go

```diff
@@ -641,6 +641,10 @@ func (a *Agent) runPostGroupCommands(ctx context.Context, tc *taskContext) {
 
 // runEndTaskSync runs task sync if it was requested for the end of this task.
 func (a *Agent) runEndTaskSync(ctx context.Context, tc *taskContext, detail *apimodels.TaskEndDetail) {
+	if tc.taskModel == nil {
+		tc.logger.Task().Error("Error running task sync.")
```

Comment:
Can we please make the error more specific to the fact that the task model being nil?

Reply:
done